### PR TITLE
PR for primus/megatron v25.7 release

### PR DESCRIPTION
**Pull request by vidushi8**
This PR updates the training yamls and adds qwen models:

1. Update DSV2, llama3, llama2 yamls.
2. Add DSV3 pretrain yaml
3. Add /Qwen2.5-7b/Qwen2.5-72B model and pretrain yamls

Reference Performance [sheet](https://amdcloud.sharepoint.com/:x:/r/sites/AIG/AIMA/Shared%20Documents/Training%20Performance/Megatron-LM%20Perf/[Pytorch_Megatron_unified_docker.xlsx](https://amdcloud.sharepoint.com/:x:/r/sites/AIG/AIMA/Shared%20Documents/Training%20Performance/Megatron-LM%20Perf/Pytorch_Megatron_unified_docker.xlsx?d=w1a6df9801626434b891ab1d06e7e60fa&csf=1&web=1&e=IaFflg&nav=MTVfezg2Q0E1NERGLTJFNTctNDA4OS1BNDg3LThDOTBCRDAzMjFENX0)?d=w1a6df9801626434b891ab1d06e7e60fa&csf=1&web=1&e=IaFflg&nav=MTVfezg2Q0E1NERGLTJFNTctNDA4OS1BNDg3LThDOTBCRDAzMjFENX0)

### CODE REVIEW

File: examples/megatron/configs/deepseek_v2-pretrain.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ modules:
       # 20250321: need latest megatron docker image
       moe_permute_fusion: false
       # fused wgrad gemm and accumulation
-      gradient_accumulation_fusion: true
+      gradient_accumulation_fusion: false
```
Comment (wenxie-amd) [inline examples/megatron/configs/deepseek_v2-pretrain.yaml:63]: I noticed that gradient accumulation fusion is disabled in this change. Could you share the reason for turning it off? Is it due to a performance issue or a compatibility concern?

File: examples/megatron/configs/llama3.3_70B-pretrain.yaml [hunk 2]
```diff
@@ -23,6 +23,8 @@ modules:
       seq_length: 8192
       max_position_embeddings: 8192
 
+      tokenizer_model: meta-llama/Llama-3.3-70B-Instruct
```
Comment (wenxie-amd) [inline examples/megatron/configs/llama3.3_70B-pretrain.yaml]: Move the tokenizer config to the model yaml (e.g. create a Llama3.3_70B.yaml, based on Llama3_70B.yaml)

File: examples/megatron/configs/qwen2.5_72B-pretrain.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+work_group: ${TEAM:amd}
+user_name: ${USER:root}
+exp_name: ${EXP_NAME:qwen2.5_72B-pretrain}
+workspace: ./output
+
+modules:
+  pre_trainer:
+    framework: megatron
+    config: pre_trainer.yaml
+    # data_path: ./data
```
Comment (wenxie-amd) [inline examples/megatron/configs/qwen2.5_72B-pretrain.yaml]: remove this comment

File: examples/megatron/configs/deepseek_v2-pretrain.yaml (hunk 1 above)
Comment (vidushi8) [reply examples/megatron/configs/deepseek_v2-pretrain.yaml:63]: Yes, It resolves both performance and compatibility issues. 

If the gradient_accumulation_fusion is unchanged (true), then we had to turn off overlap_grad_reduce: false && overlap_param_gather: false for compatibility but the perf regression still remained.

File: examples/megatron/configs/llama3.3_70B-pretrain.yaml (hunk 2 above)
Comment (vidushi8) [reply examples/megatron/configs/llama3.3_70B-pretrain.yaml]: Done.

File: examples/megatron/configs/qwen2.5_72B-pretrain.yaml (hunk 3 above)
Comment (vidushi8) [reply examples/megatron/configs/qwen2.5_72B-pretrain.yaml]: Done.

File: examples/megatron/configs/deepseek_v2_lite-pretrain.yaml [hunk 4]
```diff
@@ -55,13 +55,15 @@ modules:
...
-      gradient_accumulation_fusion: true
+      gradient_accumulation_fusion: false
       # recommend set `false` in fp8
       moe_use_legacy_grouped_gemm: true
       # fused topk router with aux score
       moe_use_fused_router_with_aux_score: false
       # pad 192/128 for deepseek attention
       fused_padded_mla_attention: false
+      
+      multi_latent_attention: false
```
Comment (wenxie-amd) [inline examples/megatron/configs/deepseek_v2_lite-pretrain.yaml:66]: Why disable MLA of deepseek?

**Review by Xiaoming-AMD — APPROVED** [review]

File: examples/megatron/configs/deepseek_v2_lite-pretrain.yaml (hunk 4 above)
Comment (vidushi8) [reply examples/megatron/configs/deepseek_v2_lite-pretrain.yaml:66]: This is for performance uplift to match the deepseek-v2 lite TGS reported here:
https://www.amd.com/en/developer/resources/rocm-hub/dev-ai/performance-results.html#tabs-a8deaeb413-item-21cea50186-tab
